packages/user-operation-controller/src/UserOperationController.ts

```diff
@@ -537,6 +537,15 @@ export class UserOperationController extends BaseController<
     validateUpdateUserOperationResponse(response);
 
     userOperation.paymasterAndData = response.paymasterAndData ?? EMPTY_BYTES;
+    if (response.callGasLimit) {
```

Comment:
@matthewwalsh0 what kind of inconsistency do you have in mind?

To be more consistent with a "patch" operation, we should be able to just apply the patch object like `{ ...userOp, ...patch }` (after validating the [reponse](https://github.com/MetaMask/keyring-api/blob/main/src/eth/erc4337/types.ts#L67-L72)).

Reply:
Both inconsistency with the prepare request where they have to provide all the gas or none at all, but also inconsistency between the individual gas values in case they provide a specific gas value at the patch phase which is incompatible with an alternate gas value from prepare.

If this is ultimately to facilitate updated estimates, then it seems logical to assume they'd have all three values again so we can be confident they're all valid.